parquet/src/test/java/org/apache/iceberg/parquet/TestDictionaryRowGroupFilter.java

```diff
@@ -223,52 +219,33 @@ public void createInputFile() throws IOException {
     dictionaryStore = reader.getNextDictionaryReader();
   }
 
-  @Test
+  @TestTemplate
   public void testAssumptions() {
     // this case validates that other cases don't need to test expressions with null literals.
-    TestHelpers.assertThrows(
-        "Should reject null literal in equal expression",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> equal("col", null));
-    TestHelpers.assertThrows(
-        "Should reject null literal in notEqual expression",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> notEqual("col", null));
-    TestHelpers.assertThrows(
-        "Should reject null literal in lessThan expression",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> lessThan("col", null));
-    TestHelpers.assertThrows(
-        "Should reject null literal in lessThanOrEqual expression",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> lessThanOrEqual("col", null));
-    TestHelpers.assertThrows(
-        "Should reject null literal in greaterThan expression",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> greaterThan("col", null));
-    TestHelpers.assertThrows(
-        "Should reject null literal in greaterThanOrEqual expression",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> greaterThanOrEqual("col", null));
-    TestHelpers.assertThrows(
-        "Should reject null literal in startsWith expression",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> startsWith("col", null));
-    TestHelpers.assertThrows(
-        "Should reject null literal in notStartsWith expression",
-        NullPointerException.class,
-        "Cannot create expression literal from null",
-        () -> notStartsWith("col", null));
+    Assertions.assertThatThrownBy(() -> equal("col", null))
+        .isInstanceOf(NullPointerException.class)
+        .hasMessageContaining("Cannot create expression literal from null");
+    Assertions.assertThatThrownBy(() -> notEqual("col", null))
+        .isInstanceOf(NullPointerException.class)
+        .hasMessageContaining("Cannot create expression literal from null");
+    Assertions.assertThatThrownBy(() -> lessThan("col", null))
+        .isInstanceOf(NullPointerException.class)
+        .hasMessageContaining("Cannot create expression literal from null");
+    Assertions.assertThatThrownBy(() -> lessThanOrEqual("col", null))
```

Comment:
Hi @nastra, do you want me to revert also these changes not strictly related to #9210?

Reply:
technically these changes are not related to #9210 but I'm ok keeping those